gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/util/QueryExpressionUtil.java

```diff
@@ -148,4 +149,21 @@ public static long alignToPeriodBoundary(long timeMillis, long periodSecs, boole
         ? timeMillis
         : alignToNext ? timeMillis + (periodMillis - delta) : timeMillis - delta;
   }
+
+  public static Expression convertLongToIsoFormat(Expression expression) {
```

Comment:
```suggestion
  public static Expression tryNormalizeLiteralToIsoDurationString(Expression expression) {
```

Or something like that. There's a few different things here.

- We're still assuming it's seconds. I think that's fine, especially since it's deprecated, but if we leave "long" in the name it would be better represented by something like "literalSeconds"
- We want to accept either an isoDuration literal or a seconds literal, so I'm dropping the earlier point and just going with "normalizeLiteral"
- If we don't get a literal expression, we're falling back and returning as is - so I added "try" to the name. Better yet would be to take @kotharironak 's suggestion and make sure it's a literal before passing in - then we could treat a non-literal as an error condition - easier to catch bugs.

Reply:
what about `convertLiteralExpressionToIsoFormat` ?